diff --git a/apexd/apexd.cpp b/apexd/apexd.cpp
index 73cf5c59..ee6a2f77 100644
--- a/apexd/apexd.cpp
+++ b/apexd/apexd.cpp
@@ -2689,7 +2689,6 @@ std::vector<ApexFileRef> SelectApexForActivation(
       LOG(FATAL) << "Unexpectedly found more than two versions or none for "
                     "APEX package "
                  << package_name;
-      continue;
     }
 
     // The package must have a pre-installed version before we consider it for
