diff --git a/config/common.mk b/config/common.mk
index 5265427b..f9693971 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -140,11 +140,15 @@ PRODUCT_PACKAGES += \
 endif
 
 PRODUCT_PACKAGES += \
-    LineageSettingsProvider \
+    LineageSettingsProvider
+
+ifneq ($(TARGET_DISABLE_LINEAGE_UPDATER),true)
+PRODUCT_PACKAGES += \
     Updater
 
 PRODUCT_COPY_FILES += \
     vendor/lineage/prebuilt/common/etc/init/init.lineage-updater.rc:$(TARGET_COPY_OUT_SYSTEM_EXT)/etc/init/init.lineage-updater.rc
+endif
 
 # Config
 PRODUCT_PACKAGES += \
diff --git a/config/device_framework_matrix.xml b/config/device_framework_matrix.xml
index a0a11316..c4e02cae 100644
--- a/config/device_framework_matrix.xml
+++ b/config/device_framework_matrix.xml
@@ -1,4 +1,4 @@
-<?xml version="2.0" encoding="UTF-8"?>
+<?xml version="1.0" encoding="UTF-8"?>
 <!--
      Copyright (C) 2021-2025 The LineageOS Project
      SPDX-License-Identifier: Apache-2.0
